### PR TITLE
Onnx GraphSurgeon update graph to fold identity constant

### PR DESCRIPTION
**Pull request by inisis**
Hi, for pytorch 2.0, by using torch.onnx.export, constant with same value and shape will be shared and creates many identity nodes, these nodes should and can be folded. This pr fix this.

image

model link      
[resnet18.onnx](http://120.224.26.73:15030/aifarm/tmp/test_torchvision[resnet18].onnx)

### CODE REVIEW

File: tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py [hunk 1]
```diff
@@ -944,6 +944,13 @@ def should_eval_foldable(tensor):
             graph_tensors = self.tensors()
             for name, values in constant_values.items():
                 tensor = graph_tensors[name]
+
+                for node in tensor.outputs:
+                    if node.op == "Identity":
+                        t = node.outputs[0]
+                        t.to_constant(tensor._values)  # Using ._values avoids copying
+                        t.inputs.clear() # Constants do not need inputs
+
```
Comment (pranavm-nvidia) [inline tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py:953]: The constant-folding pass should fold away any nodes which have only constant inputs - Identity should qualify so I don't think a special pass is needed. Indeed when I run it locally without this change, the Identity nodes are folded away.

Comment (inisis) [reply tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py:953]: Hi. this happens when the following code fail

https://github.com/NVIDIA/TensorRT/blob/release/8.6/tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py#L860-L881

the minimun env is using the model that I uploaded, and install onnx==1.14.0 onnxruntime==1.14.1

```
import onnx
import onnx_graphsurgeon as gs

model = onnx.load('resnet18.onnx')
graph = gs.import_onnx(model)
graph.fold_constants().cleanup()

for onnx_node in graph.nodes:
    if onnx_node.op == "Identity":
        raise

model = gs.export_onnx(graph)
onnx.save(model, 'resnet18_gs.onnx')
```

Comment (pranavm-nvidia) [reply tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py:953]: Ah I see - the error is: 
```
[ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Failed to load model with error: /onnxruntime_src/onnxruntime/core/graph/model.cc:146 onnxruntime::Model::Model(onnx::ModelProto&&, const PathString&, const IOnnxRuntimeOpSchemaRegistryList*, const onnxruntime::logging::Logger&, const onnxruntime::ModelOptions&) Unsupported model IR version: 9, max supported IR version: 8
```
The issue is that `onnxruntime` doesn't support IR 9. Can you try downgrading the `onnx` package to a version that uses IR 8? For example, if I install `onnx==1.13.1`, then the error goes away.

Comment (inisis) [reply tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py:953]: Yes, down grade onnx and onnxruntime version can solve this problem. I wonder if we can handle it without runtime.

Comment (pranavm-nvidia) [reply tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py:953]: We could, but then you can make the same argument for any other operator and we'd end up reimplementing all of `onnxruntime`. Is there a reason that `Identity` specifically needs to be folded? I would have thought it's the least important to fold since it's a no-op.

Comment (inisis) [reply tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py:953]: I happened to meet this case. For tools like onnx-simplifier, it has special pass to optimize this case, so I think onnx graphsurgeon should support it.

Comment (inisis) [reply tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py:953]: I wrote a onnx compiler myself, and it relies heavily on onnx graphsugeon for graph optimization, if this identity can not be folded, the compiler will fail, so if this pr can noe be merged, I will save a copy myself. Thank you for you patience.

Comment (pranavm-nvidia) [reply tools/onnx-graphsurgeon/onnx_graphsurgeon/ir/graph.py:953]: The way I see it is this:
- When `onnx`/`onnxruntime` are setup correctly, `Identity` nodes are eliminated as expected. 
- Otherwise, very little constant-folding takes place (mainly just `Shape` and related ops) because `onnx_graphsurgeon` does not include implementations for ONNX operators. I don't think it should either since it would be difficult to maintain and wasteful given that `onnxruntime` exists already.

If we were to introduce special passes that don't rely on `onnxruntime`, `Identity` would probably be lowest on the priority list since like I mentioned before, it's a no-op. Eliminating it provides no direct benefit that I can think of beyond making the `Netron` view a little nicer. 

In short, I'd recommend just making sure that `onnxruntime` and `onnx` are setup correctly, and the `Identity` nodes should be folded as you want. It would be nice if there was a way to query `onnxruntime` to determine which IR (and consequently `onnx` package) versions it supports, but I don't think that exists currently. In the meantime, there is a [support matrix](https://onnxruntime.ai/docs/reference/compatibility.html#onnx-opset-support) provided in the documentation.
